Source/JavaScriptCore/wasm/WasmIPIntGenerator.h

```diff
@@ -38,6 +39,168 @@ struct ModuleInformation;
+    uint16_t toPop; // 2B for stack values to keep
+    uint16_t toKeep;
+};
+
+struct MDBranch {
+    MDBranchTarget target;
+    MDInstructionLength instructionLength;
+};
+
+struct MDSwitch {
+    uint32_t size; // 4B for number of jump targets
+    MDBranchTarget target[0];
+};
+
+// Global get/set metadata structure
+
+struct MDGlobal {
+    uint32_t index; // 4B for index of global
+    MDInstructionLength instructionLength;
+    uint8_t bindingMode; // 1B for bindingMode
+    uint8_t isRef; // 1B for ref flag
+};
+
+// Constant metadat structures
+
+struct MDConst32 {
+    MDInstructionLength instructionLength;
+    uint32_t value;
+};
+
+struct MDConst64 {
+    MDInstructionLength instructionLength;
+    uint64_t value;
+};
+
+struct MDConst128 {
+    MDInstructionLength instructionLength;
+    v128_t value;
+};
+
+struct MDTableInit {
+    uint32_t elementIndex; // 4B for index of element
+    uint32_t tableIndex; // 4B for index of table
+    uint32_t dst; // 4B for destination (set by interpreter)
+    uint32_t src; // 4B for source (set by interpreter)
+    uint32_t length; // 4B for length (set by interpreter)
+    MDInstructionLength instructionLength;
+};
+
+struct MDTableFill {
+    uint32_t tableIndex; // 4B for index of table
+    EncodedJSValue fill; // 8B for fill value
+    uint32_t offset; // 4B for offset (set by interpreter)
+    uint32_t length; // 4B for length (set by interpreter)
+    MDInstructionLength instructionLength;
+};
+
+struct MDTableGrow {
+    uint32_t tableIndex; // 4B for index of table
+    EncodedJSValue fill; // 8B for fill value
+    uint32_t length; // 4B for length (set by interpreter)
+    MDInstructionLength instructionLength;
+};
+
+struct MDTableCopy {
+    uint32_t dstTableIndex; // 4B for index of destination table
+    uint32_t srcTableIndex; // 4B for index of source table
+    uint32_t dstOffset; // 4B for destination offset (set by interpreter)
+    uint32_t srcOffset; // 4B for source offset (set by interpreter)
+    uint32_t length; // 4B for length (set by interpreter)
+    MDInstructionLength instructionLength;
+};
+
+// Metadata structure for calls:
+
+struct MDCallEntry {
+    JSWebAssemblyInstance* instance; // Ptr for callee instance (set by operation)
+    void* entrypoint; // Ptr for callee entrypoint (set by operation)
+    EncodedJSValue boxedCallee; // 8B for callee (set by operation)
+};
+
+struct MDCall {
+    uint8_t length; // 1B for instruction length
+    uint32_t functionIndex; // 4B for decoded index
+    MDCallEntry callee;
+};
+
+struct MDCallIndirect {
+    uint8_t length; // 1B for length
+    uint32_t tableIndex; // 4B for table index
+    uint32_t typeIndex; // 4B for type index
+    uint32_t functionRef; // 4B for function ref (set by interpreter)
+    CallFrame* callFrame; // Ptr for call frame (set by interpreter)
+    MDCallEntry callee;
+};
+
+enum class MINTCall : uint8_t {
+    A = 0x0, // 0x00 - 0x07: push into a0, a1, ...
```

Comment:
Could we rename the enum cases for `A` and `Fa` as well? Knowing the LLInt assembly, they make sense, but without that knowledge it may be trickier to understand.